R/dockerfile.R

```diff
@@ -27,38 +19,130 @@
   } else if(inherits(x = from, "character") && dir.exists(from)) {
-    .dockerfile <- dockerfileFromWorkspace(path = from, to)
+    .dockerfile <- dockerfileFromWorkspace(path = from, .dockerfile = .dockerfile)
     .originalFrom <- from
-  } else {
+  }else if(is.null(from)) {
+      #Creates a basic dockerfile without the 'from'-argument 
+  }else {
     stop("Unsupported 'from': ", class(from), from)
   }
-
-  flog.info("Created Dockerfile at %s based on %s", to, .originalFrom)
-  message("Created Dockerfile at", to, "based on", .originalFrom)
+  
+  flog.info("Created Dockerfile-Object based on %s", .originalFrom)
+  message("Created Dockerfile-Object based on ", .originalFrom, ".")
   return(.dockerfile)
 }
 
+#' Format a dockerfile object to a series of instruction
+#'
+#' @param x An object of class Dockerfile
+#' @param ... Arguments to be passed down to format.default
+#'
+#' @return The Content of the dockerfile represented by the dockerfile object, by default formated as a List of strings where each string represent a new line
+#' @export
+#'
+#' @examples
+#' format(dockerfile())
+format.Dockerfile <- function(x, ...){
+  #initialize dockerfile with from
+  output = list()
+  from = toString(slot(x,"image"))
+  output = append(output, from)
+  maintainer = slot(x,"maintainer")
+  if(!is.null(maintainer))
+    output = append(output, toString(maintainer))
+  instructions = slot(x,"instructions")
+  if(!is.null(instructions) && length(instructions) > 0){
+    instructions = sapply(instructions, toString)
+    output = append(output, unlist(instructions))
+  }
+  cmd = slot(x,"cmd")
+  if(!is.null(cmd))
+    output = append(output, toString(cmd))
+  return(format(output, ...))
+}
+
+
+
+write.Dockerfile = function(x, file = paste0(getwd(), "/", "Dockerfile")){
+  flog.info("Writing dockerfile to %s", file)
+  message("Writing dockerfile to ", file)
+  return(write(as.character(format(x)), file))
+}
+
+#' Write a dockerfile
+#'
+#' @param x Dockerfile-object to be serialized
+#' @param file optional argument for specifying a costum file path 
+#'
+#' @export
+#'
+#' @examples
+#' # write a dockerfile with default parameters to temporary file and show content:
+#' temp = tempfile()
+#' write(dockerfile(), file=temp) 
+#' print(readLines(temp)) 
+#' unlink(temp)
+setMethod("write", signature(x = "Dockerfile", file = "character"), write.Dockerfile)
+  
+ 
 
-dockerfileFromSession <- function(session, to) {
-  return(NA)
+dockerfileFromSession <- function(session, .dockerfile) {
+  return(.dockerfile)
 }
 
-dockerfileFromFile <- function(file, to) {
-  return(NA)
+dockerfileFromFile <- function(file, .dockerfile) {
+  return(.dockerfile)
 }
 
-dockerfileFromWorkspace <- function(path, to) {
+dockerfileFromWorkspace <- function(path, .dockerfile) {
   .rFiles <- dir(path = path, pattern = "\\.R$", full.names = TRUE, include.dirs = FALSE, recursive = TRUE)
 
-  return(NA)
+  return(.dockerfile)
+}
+
+tagsfromRemoteImage = function(image){
```

Comment:
I like this sanity check here! 

However, the package should still work if there is no internet connection! So please handle this gracefully.
If there is no internet connection, then just assume the image does have the tag as specified by the user. If there is no such tag, the user will get an error message when ze builds the image.